src/Stratis.Features.FederatedPeg.IntegrationTests/NodeInitialisationTests.cs

```diff
@@ -109,36 +110,50 @@ public void MainChainGatewayStarts()
         [Fact]
         public void MinerPairStarts()
         {
-            using (SidechainNodeBuilder sideNodeBuilder = SidechainNodeBuilder.CreateSidechainNodeBuilder(this))
-            using (NodeBuilder nodeBuilder = NodeBuilder.Create(this))
+            try
             {
-                CoreNode main = nodeBuilder.CreateStratisPosNode(this.mainNetwork).WithWallet();
-                main.AppendToConfig("addressindex=1");
+                //Use a custom sidechain network just for this test.
+                NetworkRegistration.Clear();
+                CirrusRegTest collateralSidechainNetwork = (CirrusRegTest)CirrusNetwork.NetworksSelector.Regtest();
```

Comment:
@codingupastorm in our other integration tests, where we mess with network related settings, we create a custom network which inherits from the network we want to alter:

```
    public sealed class BitcoinRegTestNoValidationRules : BitcoinRegTest
    {
        public BitcoinRegTestNoValidationRules()
        {
            this.Name = Guid.NewGuid().ToString();
        }
    }
```

Its cleaner and less hacky than this... can you implement it this way?

Reply:
The Cirrus constructors are internal, I figured we didn't want to change that. I can do that instead though.

Reply:
👍

Reply:
donezo